### PR TITLE
Optionally emit a navigation event when a link to the current page is clicked

### PR DESCRIPTION
**Pull request by AdamNiederer**
This is enabled by setting `:reload-same-path?` to true in when calling `configure-navigation!`

### CODE REVIEW

**Comment by iku000888** [comment]
Thanks! Could you provide some example usages?

**Comment by AdamNiederer** [comment]
I've added some documentation to the README. My current use case is making all links scroll the browser to the top of the page via the nav handler (this doesn't work on links to the current page on master). I'm assuming this behaviour would be useful with other types of navigation hooks as well, or perhaps for resetting the current page's state via a link.

**Comment by iku000888** [comment]
Thanks for the explanation! I think it is an option worth adding. 
Lemme test this patch in one of my projects...

**Comment by iku000888** [comment]
Can confirm the patch works and breaks nothing in my project.
@venantius I think this would be a good enhancement. Thoughts?

**Comment by venantius** [comment]
This seems like a reasonable request to me. One clarifying question, just to make sure I'm understanding - clicking the same link won't trigger a full reload, right? It'll just emit an event?

**Comment by AdamNiederer** [comment]
> This seems like a reasonable request to me. One clarifying question, just to make sure I'm understanding - clicking the same link won't trigger a full reload, right? It'll just emit an event?

That's correct

File: README.md [hunk 1]
```diff
@@ -69,6 +69,16 @@ Using bidi, path-exists? would look like:
   (boolean (bidi/match-route app-routes path)))
 ```
 
+By default, clicking a link to the currently active route will not run the
```
Comment (venantius) [inline README.md]: I'd prefer "trigger" to "run" both here and on line 73.

File: src/accountant/core.cljs [hunk 2]
```diff
@@ -98,10 +98,13 @@
...
          (when (not= current-relative-href relative-href) ;; do not add duplicate html5 history state
            (. history (setToken relative-href title)))
-         (.preventDefault e))))))
+         (.preventDefault e)
+         (when reload-same-path?
+           (events/dispatchEvent history (Event. path true))))))))
 
 (defonce nav-handler nil)
 (defonce path-exists? nil)
+(defonce reload-same-path? nil)
```
Comment (venantius) [inline src/accountant/core.cljs]: I don't think we need this?

File: src/accountant/core.cljs [hunk 3]
```diff
@@ -111,14 +114,15 @@
...
 
   path-exists?: a fn of one argument, a path. Return truthy if this path is handled by the SPA"
-  [{:keys [nav-handler path-exists?]}]
+  [{:keys [nav-handler path-exists? reload-same-path?]}]
   (.setUseFragment history false)
   (.setPathPrefix history "")
   (.setEnabled history true)
   (set! accountant.core/nav-handler nav-handler)
   (set! accountant.core/path-exists? path-exists?)
+  (set! accountant.core/reload-same-path? path-exists?)
```
Comment (venantius) [inline src/accountant/core.cljs]: I also don't think we need this, and if we do I don't understand why `path-exists?` would be the argument to it.

File: README.md (hunk 1 above)
Comment (AdamNiederer) [reply README.md]: Thanks for the feedback - I've addressed each comment in the above two commits

**Comment by AdamNiederer** [comment]
Thanks for the feedback - I've addressed each comment in the above commits.

**Comment by venantius** [comment]
LGTM! Thanks :)

**Pull request merged**